src/main/java/core/basesyntax/strategy/PurchaseOperation.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.ShopOperation;
+import exception.OperationException;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void handle(ShopOperation shopOperation) {
+        String fruit = shopOperation.getFruit();
+        int quantity = Storage.fruitsStorage.get(fruit);
```

Comment:
Consider adding a check to handle cases where the fruit is not present in the `fruitsStorage`. This will prevent potential `NullPointerException` if a purchase operation is attempted for a fruit not in storage.